engine/table/src/main/java/io/deephaven/engine/table/impl/by/PartitionByChunkedOperator.java

```diff
@@ -477,19 +470,46 @@ private boolean appendShifts(@NotNull final LongChunk<? extends RowKeys> preShif
         }
         if (builder == null) {
             final RowSet tableRowSet = tables.getUnsafe(destination).getRowSet();
-            final RowSetBuilderRandom removedRowSetBuilder = removedBuilders.getUnsafe(destination);
+            final RowSetBuilderRandom removedRowSetBuilder =
+                    (RowSetBuilderRandom) removedBuilders.getUnsafe(destination);
             final RowSet preShiftKeys;
             if (removedRowSetBuilder == null) {
                 preShiftKeys = tableRowSet.copy();
             } else {
-                // we want to keep the builder in `removedRowSetBuilders` ready for a call to `extractAndClearRowSet` so
-                // we remake the builder after this call
-                try (final RowSet rs = removedRowSetBuilder.build()) {
-                    preShiftKeys = tableRowSet.minus(rs);
-                    RowSetBuilderRandom newBuilder = RowSetFactory.builderRandom();
-                    newBuilder.addRowSet(rs);
-                    removedBuilders.set(destination, newBuilder);
+                // dummy builder to allow a subsequent call to build()`
```

Comment:
I think `io.deephaven.engine.table.impl.util` would be a good package to put this in as a top-level class. We have another "dummy" builder class in the chunked agg helper, and will soon have yet another when I next merge, but maybe I'll follow your lead.